stackslib/src/net/api/getattachmentsinv.rs

```diff
@@ -0,0 +1,269 @@
+                        if let Ok(page_index) = entry.parse::<u32>() {
+                            page_indexes.insert(page_index);
+                        }
+                    }
+                }
+            }
+        }
+
+        let index_block_hash = if let Some(ibh) = index_block_hash {
+            ibh
+        } else {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expecting index_block_hash".to_string(),
+            ));
+        };
+
+        if page_indexes.is_empty() {
+            return Err(Error::DecodeError(
+                "Invalid Http request: expecting pages_indexes".to_string(),
+            ));
+        }
+
+        let mut page_index_list: Vec<u32> = page_indexes.into_iter().collect();
+        page_index_list.sort();
+
+        self.index_block_hash = Some(index_block_hash);
+        self.page_indexes = Some(page_index_list);
+
+        Ok(HttpRequestContents::new().query_string(query))
+    }
+}
+
+impl RPCRequestHandler for RPCGetAttachmentsInvRequestHandler {
+    /// Reset internal state
+    fn restart(&mut self) {
+        self.index_block_hash = None;
+        self.page_indexes = None;
+    }
+
+    fn try_handle_request(
+        &mut self,
+        preamble: HttpRequestPreamble,
+        _contents: HttpRequestContents,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let index_block_hash = self
+            .index_block_hash
+            .take()
+            .ok_or(NetError::SendError("Missing `index_block_hash`".into()))?;
+        let page_indexes = self
+            .page_indexes
+            .take()
+            .ok_or(NetError::SendError("Missing `page_indexes`".into()))?;
+
+        // We are receiving a list of page indexes with a chain tip hash.
+        // The amount of pages_indexes is capped by MAX_ATTACHMENT_INV_PAGES_PER_REQUEST (8)
+        // Pages sizes are controlled by the constant ATTACHMENTS_INV_PAGE_SIZE (8), which
+        // means that a `GET v2/attachments/inv` request can be requesting for a 64 bit vector
+        // at once.
+        // Since clients can be asking for non-consecutive pages indexes (1, 5_000, 10_000, ...),
+        // we will be handling each page index separately.
+        // We could also add the notion of "budget" so that a client could only get a limited number
+        // of pages when they are spanning over many blocks.
+        if page_indexes.len() > MAX_ATTACHMENT_INV_PAGES_PER_REQUEST {
+            let msg = format!(
+                "Number of attachment inv pages is limited by {} per request",
+                MAX_ATTACHMENT_INV_PAGES_PER_REQUEST
+            );
+            warn!("{}", msg);
+            return StacksHttpResponse::new_error(&preamble, &HttpBadRequest::new(msg))
+                .try_into_contents()
+                .map_err(NetError::from);
+        }
+        if page_indexes.len() == 0 {
```

Comment:
Should generally do is_empty() instead of len() == 0. Clippy will complain :P

Reply:
Fixed